src/modules/villages/village_service.ts

```diff
@@ -1,10 +1,24 @@
+import redis from '../../config/redis'
 import { Village as Entity } from './village_entity'
 import { Village as Repository } from './village_repository'
 
 export namespace Village {
-  export const findAllWithLocation = async (): Promise<Entity.ResponseFindAllWithLocation[]> => {
-    const items = await Repository.findAllWithLocation()
+  export const findAllWithLocation = async (requestQuery: Entity.RequestQuery): Promise<Entity.ResponseFindAllWithLocation[]> => {
+    let data: Entity.ResponseFindAllWithLocation[]
+    if (!await redis.get('find_all_with_location')) {
+      const items = await Repository.findAllWithLocation()
+      await redis.set('find_all_with_location', JSON.stringify(responseFindAllWithLocation(items)))
+    }
+
+    data = JSON.parse((await redis.get('find_all_with_location')))
+
+    if (requestQuery.name) data = data.filter(i => i.name.toLowerCase() === requestQuery.name.toLowerCase())
```

Comment:
filter is better in querying only, not object manipulation

Reply:
ref : https://github.com/jabardigitalservice/pikobar-pelaporan-backend/blob/master/services/map.js

Reply:
better filter query database or data caching redis ?

Reply:
jadi gini, jika query nya ada filter baiknya key redis nya ada filter by query juga contoh key `find_all_with_location_nam_filter`

Reply:
nanti jadi banyak key di redis nya yah

Reply:
wkwk siap gas